crypto/bn/bn_exp.c

```diff
@@ -1390,3 +1390,86 @@ int BN_mod_exp_simple(BIGNUM *r, const BIGNUM *a, const BIGNUM *p,
     bn_check_top(r);
     return ret;
 }
+
+/*
+ * This is a variant of modular exponentiation optimization that does
+ * parallel 2-primes exponentiation using 256-bit (AVX512VL) AVX512_IFMA ISA
+ * in 52-bit binary redundant representation.
+ * If such instructions are not available it falls back to two
+ * BN_mod_exp_mont_consttime() calls.
+ */
+int BN_mod_exp_mont_consttime_x2(BIGNUM *rr1, const BIGNUM *a1, const BIGNUM *p1,
+                                 const BIGNUM *m1, BN_MONT_CTX *in_mont1,
+                                 BIGNUM *rr2, const BIGNUM *a2, const BIGNUM *p2,
+                                 const BIGNUM *m2, BN_MONT_CTX *in_mont2,
+                                 BN_CTX *ctx)
+{
+    int ret = 0;
+
+#ifdef RSAZ_ENABLED
+    BN_MONT_CTX *mont1 = NULL;
+    BN_MONT_CTX *mont2 = NULL;
+
+    if (rsaz_avx512ifma_eligible() &&
+        ((16 == a1->top) && (16 == p1->top) && (BN_num_bits(m1) == 1024) &&
+         (16 == a2->top) && (16 == p2->top) && (BN_num_bits(m2) == 1024))) {
```

Comment:
It might look strange, but it helps to prevent accidental assignments. But even so the style in this line is not consistent, so I've changed it (here and in other places) to conform OpenSSL usage.